app/models/declaration/DocumentCategory.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.declaration
+
+class DocumentCategory(val value: String) {
```

Comment:
I think you could make it `abstract`.

Reply:
It was abstract to begin with but I needed to be able to construct one in the "Document" mapper I ended up with this 

```
.transform[DocumentCategory](value => new DocumentCategory(value), category => category.value)
```